### PR TITLE
Drop CentOS 6

### PR DESCRIPTION
**Pull request by chouquette**
### What does this PR do?

This PR bumps our build images to drop CentOS 6 support and now targets CentOS 7 for RHEL derivatives. These images also bump rust to 1.64.
This allows us to bump python-cryptography

### Motivation

This is the preliminary work described in BARX-268 to address:
* VULN-5597
* VULN-5742
* VULN-5622
* VULN-5770

### CODE REVIEW

**Review by FlorentClarret — APPROVED** [review]

File: rtloader/include/rtloader.h [hunk 1]
```diff
@@ -35,7 +35,7 @@ class RtLoader
     RtLoader(cb_memory_tracker_t memtrack_cb);
 
     //! Destructor.
-    virtual ~RtLoader(){};
+    virtual ~RtLoader() {};
```
Comment (FlorentClarret) [inline rtloader/include/rtloader.h:38]: This breaks the linters I think

Comment (chouquette) [reply rtloader/include/rtloader.h:38]: Ah, this change was meant to fix a [linter error](https://app.circleci.com/pipelines/github/DataDog/datadog-agent/127282/workflows/5c35b4a3-2ab0-4d08-bdae-842a79282d25/jobs/1452610) in circle-ci, I suppose we're having mismatching versions with different rules.
The circleci image is using `Ubuntu clang-format version 19.0.0`
while the debian image which runs most of the gitlab linting is using `clang-format version 8.0.0 (tags/RELEASE_800/final)`

File: releasenotes/notes/drop-centos6-f7afd46b4ea4cca8.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+deprecations:
+  - |
+    This release drops support for CentOS 6.
```
Comment (bkabrda) [inline releasenotes/notes/drop-centos6-f7afd46b4ea4cca8.yaml:11]: Let's also explicitly mention RHEL to be perfectly clear.

**Review by bkabrda — APPROVED** [review]
Approving with a small nit on the changelog entry.

File: rtloader/include/rtloader.h (hunk 1 above)
Comment (chouquette) [reply rtloader/include/rtloader.h:38]: Both version definitely disagree with each other:
```
root@8a0ab9905e33:/tmp/datadog-agent# clang-format -n rtloader/include/rtloader.h 
rtloader/include/rtloader.h:38:24: warning: code should be clang-formatted [-Wclang-format-violations]
    virtual ~RtLoader(){};
                       ^
```

```
 ~/dd/datadog-agent   chouquette/bump_cryptography  clang-format-15 -n rtloader/include/rtloader.h
rtloader/include/rtloader.h:38:24: warning: code should be clang-formatted [-Wclang-format-violations]
    virtual ~RtLoader() {};
                       ^
```

**Comment by chouquette** [comment]
/trigger-ci --variable RUN_ALL_BUILDS=true --variable RUN_KITCHEN_TESTS=true --variable RUN_E2E_TESTS=on --variable RUN_UNIT_TESTS=on

**Review by jhgilbert — APPROVED** [review]

**Comment by chouquette** [comment]
/trigger-ci --variable RUN_ALL_BUILDS=true --variable RUN_KITCHEN_TESTS=true --variable RUN_E2E_TESTS=on --variable RUN_UNIT_TESTS=on

**Comment by chouquette** [comment]
/trigger-ci --variable RUN_ALL_BUILDS=true --variable RUN_KITCHEN_TESTS=true --variable RUN_E2E_TESTS=on --variable RUN_UNIT_TESTS=on

**Review by Pythyu — APPROVED** [review]

**Review by pducolin — APPROVED** [review]

**Pull request merged**